product_net_weight/views/view_product_product.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<!--
+Copyright (C) 2021 - Today: GRAP (http://www.grap.coop)
+@author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+-->
+<odoo>
+
+    <record id="view_product_product_form" model="ir.ui.view">
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_variant_easy_edit_view" />
+        <field name="arch" type="xml">
+            <xpath expr="//label[@for='weight']" position="before">
+                <label for="net_weight" />
+                <div class="o_row">
+                    <field name="net_weight" />
+                    <span><field name="weight_uom_name" /></span>
```

Comment:
Would remove <span> to align unit with standard product form (and in additions below)